gtk/src/toga_gtk/widgets/numberinput.py

```diff
@@ -53,10 +53,11 @@ def set_value(self, value):
             self.native.set_value(self.interface.value)
 
     def set_alignment(self, value):
-        self.interface.factory.not_implemented("NumberInput.set_alignment()")
+        xalign, justify = gtk_alignment(value)
+        self.native.set_alignment(xalign)
 
     def set_font(self, font):
-        self.interface.factory.not_implemented("NumberInput.set_font()")
+        super().set_font(font)
```

Comment:
This is effectively a no-op - if we don't need any modifications to the base class implementation, then there's no need to define it here.

If the issue is the test failing because of a missing `set_font` method - the fix is to remove the `set_font` requirement from the dummy backend.